internal/pkg/client/gitlab/fetch_test.go

```diff
@@ -0,0 +1,201 @@
+) (*gogitlab.DependencyListExport, *gogitlab.Response, error) {
+	args := mc.Called(pipelineID, opt, options)
+
+	return args.Get(0).(*gogitlab.DependencyListExport), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck,lll
+}
+
+func (mc *mockClient) GetDependencyListExport(
+	id int,
+	options ...gogitlab.RequestOptionFunc,
+) (*gogitlab.DependencyListExport, *gogitlab.Response, error) {
+	args := mc.Called(id, options)
+
+	return args.Get(0).(*gogitlab.DependencyListExport), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck,lll
+}
+
+func (mc *mockClient) DownloadDependencyListExport(
+	id int,
+	options ...gogitlab.RequestOptionFunc,
+) (io.Reader, *gogitlab.Response, error) {
+	args := mc.Called(id, options)
+
+	return args.Get(0).(io.Reader), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck
+}
+
+var successGitLabResponse = &gogitlab.Response{
+	Response: &http.Response{
+		Status:     "200 OK",
+		StatusCode: 200,
+		Proto:      "HTTP/1.0",
+	},
+}
+
+func TestCreateExport(t *testing.T) { //nolint:paralleltest
+	mockedGoGitLabClient := &mockClient{}
+
+	dummyProjectID := 1234
+	dummyProjectName := "DUMMY_PROJECT"
+	dummyBranchName := "DUMMY_BRANCH"
+	dummyCommitSHA := "ABCD"
+	dummyPipelineID := 2345
+
+	expectedDependencyListExport := &gogitlab.DependencyListExport{
+		ID:          3456,
+		HasFinished: false,
+		Self:        "TEST",
+		Download:    "TEST/Download",
+	}
+
+	mockedGoGitLabClient.On(
+		"GetProject",
+		dummyProjectName,
+		(*gogitlab.GetProjectOptions)(nil),
+		[]gogitlab.RequestOptionFunc(nil),
+	).Return(
+		&gogitlab.Project{
+			ID:   dummyProjectID,
+			Name: dummyProjectName,
+		},
+		successGitLabResponse,
+		nil,
+	)
+
+	mockedGoGitLabClient.On(
+		"GetBranch",
+		dummyProjectID,
+		dummyBranchName,
+		[]gogitlab.RequestOptionFunc(nil),
+	).Return(
+		&gogitlab.Branch{
+			Name: dummyBranchName,
+			Commit: &gogitlab.Commit{
+				ID: dummyCommitSHA,
+			},
+		},
+		successGitLabResponse,
+		nil,
+	)
+
+	mockedGoGitLabClient.On(
+		"GetCommit",
+		dummyProjectID,
+		dummyCommitSHA,
+		(*gogitlab.GetCommitOptions)(nil),
+		[]gogitlab.RequestOptionFunc(nil),
+	).Return(
+		&gogitlab.Commit{
+			ID: dummyCommitSHA,
+			LastPipeline: &gogitlab.PipelineInfo{
+				ID: dummyPipelineID,
+			},
+		},
+		successGitLabResponse,
+		nil,
+	)
+
+	mockedGoGitLabClient.On(
+		"CreateDependencyListExport",
+		dummyPipelineID,
+		(*gogitlab.CreateDependencyListExportOptions)(nil),
+		[]gogitlab.RequestOptionFunc(nil),
+	).Return(
+		expectedDependencyListExport, successGitLabResponse, nil,
+	)
+
+	session := &dependencyListExportSession{
+		Client: mockedGoGitLabClient,
+	}
+
+	if err := session.createExport("DUMMY_PROJECT", "DUMMY_BRANCH"); err != nil {
+		t.Errorf("failed to create dependency list export: %v", err)
+	}
+
+	mockedGoGitLabClient.AssertExpectations(t)
+}
+
+func TestPollExportUntilFinished(t *testing.T) { //nolint:paralleltest
```

Comment:
Please avoid ignoring linters. Use a testify suite as suggested above and it won't be needed (need to fix multiple instances)